dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2763,8 +2768,27 @@ void MergeTreeData::movePartitionToVolume(const ASTPtr & partition, const String
             if (part->disk->getName() == disk->getName())
                 throw Exception("Part " + part->name + " already on volume '" + name + "'", ErrorCodes::UNKNOWN_DISK);
 
-    if (!movePartsToSpace(parts, std::static_pointer_cast<const DiskSpace::Space>(volume)))
-        throw Exception("Cannot move parts because moves are manually disabled.", ErrorCodes::ABORTED);
+    parts.erase(std::remove_if(parts.begin(), parts.end(), [&](auto part_ptr) {
+        for (const auto & disk : volume->disks)
+        {
+            if (part_ptr->disk->getName() == disk->getName())
+            {
+                return true;
+            }
+        }
+        return false;
+    }), parts.end());
+
+
+    if (!parts.empty())
+    {
+        if (!movePartsToSpace(parts, std::static_pointer_cast<const DiskSpace::Space>(volume)))
+            throw Exception("Cannot move parts because moves are manually disabled.", ErrorCodes::ABORTED);
+    }
+    else
+    {
+        LOG_DEBUG(log, "No parts of partition " << partition_id << " to move to volume " << volume->getName());
```

Comment:
Throw exception. User should know about it.

Reply:
3debdc211904eb0b5fe6cca1a4af99d8db635892